### PR TITLE
Split pricing dialog

### PR DESCRIPTION
**Pull request by cabch**
needed to make the dialog refactoring correctly - less impact doing it this way than doing dialog refactoring before pricing split

### CODE REVIEW

File: src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      this.clearAndResetControl(this[`${event.source.id}`]);
+    }
+    this.formGroup.markAsDirty();
+    this.formGroup.updateValueAndValidity();
+  }
+
+  public toggleEnergy(event: MatSlideToggleChange) {
+    this[`${event.source.id}Enabled`].setValue(event.checked);
+    if (event.checked) {
+      this[event.source.id].setValidators(Validators.compose([
```
Comment (ClaudeROSSI) [inline src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts:167]: idem ... not called when editing

File: src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import PricingDefinition from '../../../../types/Pricing';
+import { Constants } from '../../../../utils/Constants';
+import { PricingHelpers } from '../../../../utils/PricingHelpers';
+
+@Component({
+  selector: 'app-pricing-definition-restrictions',
+  templateUrl: './pricing-definition-restrictions.component.html',
+})
+
+export class PricingDefinitionRestricitionsComponent implements OnInit, OnChanges {
```
Comment (ClaudeROSSI) [inline src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts:16]: typo - class name is wrong "restric**i**tions"

image

File: src/app/utils/PricingHelpers.ts [hunk 3]
```diff
@@ -18,4 +20,20 @@ export class PricingHelpers {
   public static convertDurationToSeconds( enabled: boolean, value: number): number {
     return (enabled)? PricingHelpers.toSeconds(value): null;
   }
+
+  public static minMaxValidator(formGroup: FormGroup, minControl: string, maxControl: string): ValidatorFn {
+    const restrictions = formGroup.controls.restrictions as FormGroup;
+    return (control: AbstractControl): { [key: string]: any } | null => {
+      const fieldMinValue = restrictions.controls[minControl]?.value;
+      const fieldMaxValue = restrictions.controls[maxControl]?.value;
+      if (!Utils.isNullOrUndefined(fieldMinValue) && !Utils.isNullOrUndefined(fieldMaxValue) && fieldMaxValue !== '' && fieldMinValue !== '' && fieldMinValue >= 0 && fieldMaxValue >= 0) {
```
Comment (ClaudeROSSI) [inline src/app/utils/PricingHelpers.ts:29]: ??? why do we check for empty strings? 
??? fieldMinValue !== '' - I do not understand!

File: src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  }
+
+  public toggleTimeRange(event: MatSlideToggleChange) {
+    this.timeRangeEnabled.setValue(event.checked);
+    if (event.checked) {
+      this.timeFrom.setValidators(Validators.required);
+      this.timeTo.setValidators(Validators.required);
+    } else {
+      this.clearAndResetControl(this.timeFrom);
+      this.clearAndResetControl(this.timeTo);
```
Comment (ClaudeROSSI) [inline src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts:141]: Not called when editing a pricing definition where the restrictions are already set.

Should be called explicitly when openning the dialog!

**Review by ClaudeROSSI — CHANGES_REQUESTED** [review]
a few unexpected behaviors when editing.

File: src/app/utils/PricingHelpers.ts (hunk 3 above)
Comment (cabch) [reply src/app/utils/PricingHelpers.ts:29]: if field has been erased but still toggled on we have an empty string so it passes the if condition and goes to the Number('emptystring') that resolves to zero number whereas we should not have gone that far

Screenshot 2022-03-02 at 09 50 50
g

File: src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts (hunk 2 above)
Comment (cabch) [reply src/app/shared/pricing-definitions/pricing-definition/restrictions/pricing-definition-restrictions.component.ts:16]: oups
✅ modified in next commit

File: src/app/utils/PricingHelpers.ts (hunk 3 above)
Comment (ClaudeROSSI) [reply src/app/utils/PricingHelpers.ts]: thanks for the changes

File: src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts [hunk 5]
```diff
@@ -453,20 +141,20 @@ export class PricingDefinitionComponent implements OnInit {
...
-      timeTo: (this.timeRangeEnabled.value)? this.timeToValue.value: null,
-      minEnergyKWh: (this.minEnergyKWhEnabled.value)? this.minEnergyKWhValue.value: null,
-      maxEnergyKWh: (this.maxEnergyKWhEnabled.value)? this.maxEnergyKWhValue.value: null,
-      minDurationSecs: PricingHelpers.convertDurationToSeconds(this.minDurationEnabled.value, this.minDuration.value),
-      maxDurationSecs: PricingHelpers.convertDurationToSeconds(this.maxDurationEnabled.value, this.maxDuration.value),
+      daysOfWeek: this.pricingDefinitionRestrictions.selectedDays.value || null,
+      timeFrom: this.pricingDefinitionRestrictions.timeFrom.value || null,
+      timeTo: this.pricingDefinitionRestrictions.timeTo.value || null,
+      minEnergyKWh: this.pricingDefinitionRestrictions.minEnergyKWh.value || null,
+      maxEnergyKWh: this.pricingDefinitionRestrictions.maxEnergyKWh.value || null,
```
Comment (ClaudeROSSI) [inline src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts:155]: @cabch : As discussed, please make sure to only save the value when the toggle is ON.

**Review by ClaudeROSSI — CHANGES_REQUESTED** [review]
as discussed on the phone ... check my comments

**Review by ClaudeROSSI — APPROVED** [review]
tested and approved.
Thanks

**Comment by ClaudeROSSI** [comment]
@LucasBrazi06 : Everything is ok from my perspective.

**Review by LucasBrazi06 — APPROVED** [review]
Sounds good for me.
Feel free to change what I did after the merge.
Just one remark, if I create a pricing with only a name and a description, is it okay? Does it mean it's free?

**Pull request merged**